dali/kernels/imgproc/resample/resampling_setup.cc

```diff
@@ -98,80 +100,198 @@ SeparableResamplingSetup<spatial_ndim>::ComputeScaleAndROI(
   return roi;
 }
 
+template <int ndim>
+struct ProcessingOrderCalculator {
+  ProcessingOrderCalculator(ivec<ndim> in_size, ivec<ndim> out_size, ivec<ndim> filter_support)
+  : in_size(in_size), out_size(out_size), filter_support(filter_support) {
+  }
+
+  ProcessingOrder<ndim> operator()() {
+    for (int i = 0; i < ndim; i++)
+      best_order[i] = i;
+    min_cost = 1e+30f;
+    axis_mask = {};
+    curr_size = in_size;
+    run(0);
+    return best_order;
+  }
+
+ private:
+  float compute_cost_mul(int axis) {
+    // y-axis is likely to be the cheapest
+    return axis == 0 ? 1.4f : axis > 1 ? 1.2f : 1.0f;
+  }
+
+  static constexpr float size_bias = 3;
+
+  float calculate_cost(int pass, int axis) {
+    sizes[pass] = volume(curr_size);
+    float base_cost = filter_support[axis] * sizes[pass];
+    compute_costs[pass] = compute_cost_mul(axis) * base_cost;
+    return compute_costs[pass] + sizes[pass] * size_bias;
```

Comment:
Are the sizes and compute_cost even used outside of this function?
```suggestion
    auto vol = volume(curr_size);
    return vol * (compute_cost_mul(axis) * filter_support[axis] + size_bias);
```